### PR TITLE
Make CLI commands async

### LINKED ISSUE #872: Asyncify CLI
Needed for grpc calls to Apalace, and will help open the most ecosystem-preferred path for many JS operations. E.g., it should make it easy to implement #633.

This work will run ahead of #822, based on @konnov previous work on https://github.com/informalsystems/quint/tree/igor/sigint734

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #872

Took out parts from
https://github.com/informalsystems/quint/commit/ca7281dfe2ac435ed494bbe271db429fc4495baf

This frees us up to use async operations within the subcommands, when
needed.

- [x] Tests added for any new code
- [x] Ran `npm run format` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [-] Entries added to the respective `CHANGELOG.md` for any new functionality
- [-] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

### CODE REVIEW

File: quint/src/cliCommands.ts [hunk 1]
```diff
@@ -38,109 +43,137 @@ import { Rng, newRng } from './rng'
...
+  trace?: QuintEx[];
   /* Docstrings by defintion name by module name */
-  documentation?: Map<string, Map<string, DocumentationEntry>>,
-  errors?: ErrorMessage[],
-  warnings?: any[], // TODO it doesn't look like this is being used for anything. Should we remove it?
-  sourceCode?: string, // Should not printed, only used in formatting errors
+  documentation?: Map<string, Map<string, DocumentationEntry>>;
+  errors?: ErrorMessage[];
+  warnings?: any[]; // TODO it doesn't look like this is being used for anything. Should we remove it?
+  sourceCode?: string; // Should not printed, only used in formatting errors
```
Comment (shonfeder) [inline quint/src/cliCommands.ts:74]: This is a result of running `eslint fix` with our current config, due to defaults:

https://typescript-eslint.io/rules/member-delimiter-style/

But Apparently we aren't supposed to actually be using eslint for this?

I'm a bit confused still on the state our formatting conventions, but just hit a merge conflict due to them, so I will do a cycle to try to get this in order and under CI.

File: quint/src/cli.ts [hunk 2]
```diff
@@ -33,6 +34,21 @@ const defaultOpts = (yargs: any) =>
     type: 'string',
   })
 
+// Chain async CLIProcedures
+//
+// This saves us having to manually thread the result argument like
+//
+// `prevCmd.then(prevResult => prevResult.asyncChain(nextCmd))`
+//
+// Instaed writing:
```
Comment (konnov) [inline quint/src/cli.ts:43]:
```suggestion
// Instead writing:
```

**Review by konnov — APPROVED** [review]
Looks great!

**Review by konnov — APPROVED** [review]
Looks great!

**Pull request merged**